src/Service/ImageManager.php

```diff
@@ -128,6 +130,13 @@ public function download(string $url): ?string
         return $tempFile;
     }
 
+    public function cleanExif(string $filePath)
+    {
+        if ($this->exifCleaner->isEnabled()) {
```

Comment:
do you think it'd be cheaper if we also checked whether there still was exif data (was already scrubbed) with something like `exif_read_data()` and early out?

Reply:
I don't know much about that function or how much it can read to give an accurate answer

also it doesn't help that image metadata stuff [is quite a damn mess](https://exiftool.org/TagNames/), in particular there's [XMP tags](https://exiftool.org/TagNames/XMP.html) which seems to be a different beasts of its own and I don't know if `exif_read_data()` could read these

Reply:
Thinking about it my main concern was spending cpu cycles doing unneeded work (potentially, depending on how quick the tool is to detect it has nothing to do, but in my experience exec'ing out still has a cost). I agree that, as an instance owner, it might be nicer to have everything you cache scrubbed. Since this is opt-in, at least for the moment, instance owners that don't want to pay the cost of scrubbing all external images and choose not to enable it, I imagine